cypress/e2e/devpool.cy.ts

```diff
@@ -205,4 +205,32 @@ describe("DevPool", () => {
     cy.get("#authenticated").should("exist");
     cy.get("#filter").should("be.visible");
   });
+
+  describe("Assert that generic error modal displays", () => {
+    beforeEach(() => {
+      cy.origin("https://github.com/login", () => {
+        cy.get("#login_field").type(Cypress.env("GITHUB_USERNAME"));
+        cy.get("#password").type(Cypress.env("GITHUB_PASSWORD"));
+        cy.get(".position-relative > .btn").click();
+      });
+    });
+
+    it("should display the generic error modal if the user is not an org member with the required scope", () => {
```

Comment:
Not sure to understand this part.

If the user is not a member, then we don't show the private repositories and do not show the lock button to add access to private repos.

If the user is an org member, but didn't allow for private repos, then we display a lock icon next to his login to allow for more permission.

If the user is an org member, and allowed private access, then we just query private repos as well.

This test seems to imply errors are shown when the user is not an org member / didn't allow for private repos, which I don't think is wanted.

Reply:
> 1. There are errors in the console but the toast is not shown
> 
> 
> <img alt="Screenshot 2024-05-14 at 12 15 49" width="1512" src="https://private-user-images.githubusercontent.com/119500907/330364504-51f84ae4-e348-4fd7-824b-8ddad61268e4.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYyODE5NzIsIm5iZiI6MTcxNjI4MTY3MiwicGF0aCI6Ii8xMTk1MDA5MDcvMzMwMzY0NTA0LTUxZjg0YWU0LWUzNDgtNGZkNy04MjRiLThkZGFkNjEyNjhlNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTIxJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyMVQwODU0MzJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1hMDhmYTA5YTYyNjFjMDRhYWE5ZmZlYTA2Mzg4YmQ3NTViNTczZDdiMTY3YjQzNThlNmYzZDI1MTkxMDI2NTNjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7WnzgTtKav0PYaO_tTGjdtMAp37sXrUSXUTyzRKh_zE"> 

@gentlementlegen  this is one of the errors in the console rndquu highlighted should show a toast. So the test I wrote was to check if the err occurs in that scenario and if the toast is shown when it happens

Reply:
Also I'm currently displaying a toast for this scenario! Should I get rid of it?

Reply:
@gentlementlegen please for some feedback on this. Should I get rid of the modal on this error?

Reply:
1. Unit tests doesn't seem to work, locally I get
<img width="319" alt="image" src="https://github.com/ubiquity/work.ubq.fi/assets/9807008/22c942ac-6a2a-42d2-9597-ca42c18d009a">

2. I tried with different accounts and no modal was ever shown, which technically is what I think we want. According to your code changes it will show up only if the request itself fails or doesn't return 404, which makes sense. So please fix your test because it doesn't reflect that.
3. Please commit the other suggestions I made previously.

Reply:
@gentlementlegen worked on the e2e test and committed the previous suggestions. Everything works now!